File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const net_lost = lost.filter(elem => !reserve.includes(elem)).sort();
    const net_reserve = reserve.filter(elem => !lost.includes(elem)).sort();
    const initLost = net_lost.length;
    let lended = 0;
    
    while (net_lost.length !== 0) {
        const current = net_lost.pop();
        for (let i = net_reserve.length - 1; i >= 0; i--) {
            if (current + 1 === net_reserve[i] || current - 1 === net_reserve[i]) {
                lended++;
                net_reserve.splice(i,1)
                break;
            }
        }
    }

    var answer = n - initLost + lended;
    return answer;
}

File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    net_lost = [i for i in lost if i not in reserve]
    net_reserve = [i for i in reserve if i not in lost]
    
    net_lost.sort()
    net_reserve.sort()
    saved = 0
    p1, p2 = (0, 0)
    while p1 < len(net_lost) and p2 < len(net_reserve):
        if abs(net_lost[p1] - net_reserve[p2]) <= 1:
            p1 += 1
            p2 += 1
            saved += 1
        elif net_lost[p1] > net_reserve[p2]:
            p2 += 1
        else:
            p1 += 1
        
    return n - len(net_lost) + saved